internal/v1/handler.go

```diff
@@ -513,6 +513,32 @@ func convertIgnoreImageTypeToSlice(ignoreImageTypes *[]ImageTypes) []string {
 	return stringSlice
 }
 
+func (h *Handlers) fetchComposes(orgId string, blueprintId *uuid.UUID, blueprintVersion *int, since time.Duration, limit, offset int, ignoreImageTypeStrings []string) ([]db.ComposeEntry, int, error) {
+	if blueprintId != nil {
+		return h.server.db.GetBlueprintComposes(orgId, *blueprintId, since, limit, offset, ignoreImageTypeStrings)
+	}
+	return h.server.db.GetComposes(orgId, since, limit, offset, ignoreImageTypeStrings)
+}
```

Comment:
Ah now I understand why you just sticked the new field into the "entry" struct :-)

A new struct will work here, you just return the Compose portion of the embedded type.